src/hotspot/share/jvmci/jvmciCodeInstaller.cpp

```diff
@@ -727,6 +728,11 @@ JVMCI::CodeInstallResult CodeInstaller::install(JVMCICompiler* compiler,
       JVMCI_THROW_MSG_(IllegalArgumentException, "InstalledCode object must be a HotSpotNmethod when installing a HotSpotCompiledNmethod", JVMCI::ok);
     }
 
+    if (UseZGC && _nmethod_entry_patch_offset == -1) {
+      // ZGC requires the use of entry barriers for correctness
```

Comment:
Note that G1 also requires nmethod entry barriers for correctness.

Reply:
We've been happily running without them for a long time which is why we're just now getting around to supporting them.  Are we sitting on a ticking time bomb?  I do want to make them required as of 21 but maybe we could defer that to a future commit once my Graal changes to support the entry barriers has landed?